### ISSUE TITLE
@turf/nearest-point - explanation needed

### ISSUE THREAD

**Issue by dabajabaza** [Turfjs/turf#1101]
I use **turf-nearest-point** on array of geojson points for finding N nearest points. But it does not work as expected.  [Here is the piece of code and map with all necessary data](https://codepen.io/ckpunmep/pen/GOyWLQ).
Red dots are points that **turf-nearest-point** found as nearest to my start point (the green one).
Just visually the points 4 and 5 are nearer to point 1 than the point 2 and 3 and I measured distance on maps.yandex.ru and google.com/maps between them and it confirmed.

**Comment by DenisCarriere** [comment]
Very interesting.. this type of bug/issue sounds like the issue would fall in the distance calculation method, at the moment `@turf/nearest-point` is using `@turf/distance` to do those calculations which is a geodesic distance, maybe for the "northern" regions we could adapt a different type distance operation.

We should definitely include this in a test case to make sure it's getting the "correct" point for regions in higher latitudes.

Maybe we try using [Mapbox's Cheap Ruler](https://github.com/mapbox/cheap-ruler) to see if we get any different results in these tests.

👍 @dabajabaza Thanks for reporting this.

**Comment by rowanwins** [comment]
Hi @dabajabaza 

I've just taken a bit more of a look at this and I don't think turf is doing anything incorrect.

I simplified your test case a little and have made it available [here](https://runkit.com/rowanwins/5a3110dfbd5fe100123a4ea4). If you look at the [output](http://bl.ocks.org/d/3ad771291e0af25ed99c57594bb59328) of this it is as expected, so I suspect the error must lie in how you are handling those markers in your leaflet code.

If you could test again and close if appropriate that would be great.

Thanks,
Rowan

**Comment by dabajabaza** [comment]
@rowanwins thanks for reply. I made simplified [example ](https://codepen.io/ckpunmep/pen/YYXWyR?editors=1010) and it's work as expected.
I guess it's my bad - Leaflet and geoJSON standard use different coordinates order - [lat, lon] vs [lon, lat] and I used Array.prototype.reverse()  to switch its order that mutated original array - so the result of turf.nearesPoint() was skewed.
Sorry for wasting your time  :(
